### ISSUE TITLE
SIGSEGV segmentation violation from --help

### ISSUE THREAD

**Issue by ahmetalpbalkan** [urfave/cli#545]
Version: `v1.18.1`

go version go1.7.1 darwin/amd64

### main.go

```go
package main

import (
	"os"

	"github.com/urfave/cli"
)

func main() {
	app := &cli.App{
		Name: "hello",
		Action: func(c *cli.Context) error {
			println("Greetings")
			return nil
		},
	}
	app.Run(os.Args)
}
```

### go run main.go --help

```
panic: runtime error: invalid memory address or nil pointer dereference [recovered]
	panic: tabwriter: panic during Flush [recovered]
	panic: tabwriter: panic during Write [recovered]
	panic: tabwriter: panic during Write
[signal SIGSEGV: segmentation violation code=0x1 addr=0x20 pc=0xca969]

goroutine 1 [running]:
panic(0x111520, 0xc4200122d0)
	/usr/local/Cellar/go/1.7.1/libexec/src/runtime/panic.go:500 +0x1a1
text/template.errRecover(0xc420037c20)
	/usr/local/Cellar/go/1.7.1/libexec/src/text/template/exec.go:146 +0x18e
panic(0x111520, 0xc4200122d0)
	/usr/local/Cellar/go/1.7.1/libexec/src/runtime/panic.go:458 +0x243
text/tabwriter.handlePanic(0xc420037a28, 0x13db3c, 0x5)
	/usr/local/Cellar/go/1.7.1/libexec/src/text/tabwriter/tabwriter.go:457 +0x149
panic(0x111520, 0xc4200122c0)
	/usr/local/Cellar/go/1.7.1/libexec/src/runtime/panic.go:458 +0x243
text/tabwriter.handlePanic(0xc420037938, 0x13dac4, 0x5)
	/usr/local/Cellar/go/1.7.1/libexec/src/text/tabwriter/tabwriter.go:457 +0x149
panic(0x11d720, 0xc4200120a0)
	/usr/local/Cellar/go/1.7.1/libexec/src/runtime/panic.go:458 +0x243
text/tabwriter.(*Writer).write0(0xc4200a8100, 0xc4200a815c, 0x5, 0x40)
	/usr/local/Cellar/go/1.7.1/libexec/src/text/tabwriter/tabwriter.go:229 +0x29
text/tabwriter.(*Writer).writeLines(0xc4200a8100, 0x0, 0x0, 0x2, 0x26243)
	/usr/local/Cellar/go/1.7.1/libexec/src/text/tabwriter/tabwriter.go:296 +0x1c9
text/tabwriter.(*Writer).format(0xc4200a8100, 0x0, 0x0, 0x2, 0x5)
	/usr/local/Cellar/go/1.7.1/libexec/src/text/tabwriter/tabwriter.go:384 +0xb2
text/tabwriter.(*Writer).flush(0xc4200a8100, 0x0, 0x0)
	/usr/local/Cellar/go/1.7.1/libexec/src/text/tabwriter/tabwriter.go:483 +0x1ab
text/tabwriter.(*Writer).Flush(0xc4200a8100, 0xc4200ac2c0, 0x1)
	/usr/local/Cellar/go/1.7.1/libexec/src/text/tabwriter/tabwriter.go:466 +0x2b
text/tabwriter.(*Writer).Write(0xc4200a8100, 0xc420074200, 0x9, 0x10, 0x6, 0x0, 0x0)
	/usr/local/Cellar/go/1.7.1/libexec/src/text/tabwriter/tabwriter.go:517 +0x32d
text/template.(*state).walk(0xc420037ba0, 0x12f0a0, 0xc4200b0000, 0x16, 0x1d1b40, 0xc420076330)
	/usr/local/Cellar/go/1.7.1/libexec/src/text/template/exec.go:241 +0x51d
text/template.(*state).walk(0xc420037ba0, 0x12f0a0, 0xc4200b0000, 0x16, 0x1d18a0, 0xc420076300)
	/usr/local/Cellar/go/1.7.1/libexec/src/text/template/exec.go:234 +0x13d
text/template.(*Template).execute(0xc4200a0200, 0x1d03c0, 0xc4200a8100, 0x12f0a0, 0xc4200b0000, 0x0, 0x0)
	/usr/local/Cellar/go/1.7.1/libexec/src/text/template/exec.go:189 +0x204
text/template.(*Template).Execute(0xc4200a0200, 0x1d03c0, 0xc4200a8100, 0x12f0a0, 0xc4200b0000, 0x0, 0x0)
	/usr/local/Cellar/go/1.7.1/libexec/src/text/template/exec.go:175 +0x53
hello/vendor/github.com/urfave/cli.printHelp(0x0, 0x0, 0x14647d, 0x2ea, 0x12f0a0, 0xc4200b0000)
	/Users/alp/workspace/[redacted]/src/hello/vendor/github.com/urfave/cli/help.go:197 +0x21c
hello/vendor/github.com/urfave/cli.ShowAppHelp(0xc4200b2000, 0xc4200b2001, 0x2)
	/Users/alp/workspace/[redacted]/src/hello/vendor/github.com/urfave/cli/help.go:121 +0x78
hello/vendor/github.com/urfave/cli.(*App).Run(0xc4200b0000, 0xc42000a220, 0x2, 0x2, 0x0, 0x0)
	/Users/alp/workspace/[redacted]/src/hello/vendor/github.com/urfave/cli/app.go:198 +0x881
main.main()
	/Users/alp/workspace/[redacted]/src/hello/main.go:17 +0xc5
exit status 2
```

This is a curious case, I doubt if it's happening because I did not specified some flags or subcommands and so on. This could be why, but essentially the `text/tabwriter` probably shouldn't be doing any pointer arithmetic to cause SIGSEGV...

**Comment by ahmetalpbalkan** [comment]
This doesn't repro if I use `cli.NewApp()` instead of `&cli.App{...}` syntax.

**Comment by jszwedko** [comment]
Hi @ahmetalpbalkan,

Thank you for the detailed report and example code. This was a result of not defaulting the `Writer` of the `app` which caused `text/template` to panic when executing the help templates. It should be fixed by #547 -- can you verify that this works for you?

**Comment by jszwedko** [comment]
Let me know if #547 doesn't fix it for you
